pkg/diff/diffbycommand.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"sigs.k8s.io/yaml"
+)
+
+// DiffByCommand converts the given objects into yaml and then runs the command to compare them.
+func DiffByCommand(command string, old, new interface{}) ([]byte, error) {
```

Comment:
> [IMO]
It would be better to define it as renderByCommand.
I feel the responsibility of the method is to render the diffResult.

Similar to the reason above, how about renaming it as `RenderByCommand` and moving it in the `render.go` like below↓?
https://github.com/pipe-cd/pipecd/blob/master/pkg/diff/renderer.go#L54

I think the DiffXX methods in the diff pkg return `diff.Result` and have the responsibility to extract the differences. It is ideal to unify the interface and the responsivility as much as possible.

WDYT? I want your opinion @t-kikuc @khanhtc1202

Reply:
@ffjlabo 
Thanks, I agree with you.
TBH, when I referred to CloudRun's diff.go, I thought it was confusing and should be renamed and moved.
cf. https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/platformprovider/cloudrun/diff.go